tracing/src/macros.rs

```diff
@@ -583,39 +583,48 @@ macro_rules! error_span {
 // /// ```
 #[macro_export]
 macro_rules! event {
-    (target: $target:expr, parent: $parent:expr, $lvl:expr, { $($fields:tt)* } )=> (
-        $crate::__tracing_log!(
+    (target: $target:expr, parent: $parent:expr, $lvl:expr, { $($fields:tt)* } )=> ({
+        use $crate::__macro_support::Callsite as _;
+        static CALLSITE: $crate::__macro_support::MacroCallsite = $crate::callsite2! {
+            name: concat!(
+                "event ",
+                file!(),
+                ":",
+                line!()
+            ),
+            kind: $crate::metadata::Kind::EVENT,
             target: $target,
-            $lvl,
-            $($fields)*
-        );
+            level: $lvl,
+            fields: $($fields)*
+        };
 
-        if $crate::level_enabled!($lvl) {
-            use $crate::__macro_support::Callsite as _;
-            static CALLSITE: $crate::__macro_support::MacroCallsite = $crate::callsite2! {
-                name: concat!(
-                    "event ",
-                    file!(),
-                    ":",
-                    line!()
-                ),
-                kind: $crate::metadata::Kind::EVENT,
-                target: $target,
-                level: $lvl,
-                fields: $($fields)*
-            };
+        let enabled = $crate::level_enabled!($lvl) && {
             let interest = CALLSITE.interest();
-            if !interest.is_never() && CALLSITE.is_enabled(interest)  {
+            !interest.is_never() && CALLSITE.is_enabled(interest)
+        };
+        if enabled {
```

Comment:
i wonder if we could improve the benchmark performance for disabled spans by going back to having a separate `if` statement for `level_enabled!` and checking if the callsite is enabled? it might be worth trying that and re-running the benchmarks?

Reply:
I had a go at this, but it didn't make any difference, and the way it is now is more readable to me.

Reply:
interesting, thanks for trying it anyway!